### ISSUE TITLE
ZynAddSubFX "Show GUI" causes severe audio dropout.

### ISSUE THREAD

**Issue by mikobuntu** [LMMS/lmms#1825]
https://www.youtube.com/watch?v=IMt_oy-9ljY&feature=youtu.be

This has been a long known issue that using the " Show GUI " button on ZASF will cause a significant dropout of all audio during playback.

Hopefully there is a way to avoid this completely or at least make the transition less noticeable .

I understand the concept that ZASF runs in a Remote process .

thanks Mikobuntu :)

**Comment by Sti2nd** [comment]
So you are proposing to check if the event prevents the audio from playing cause obviously the specs of a computer would also play a role.

**Comment by mikobuntu** [comment]
@Sti2nd  Yeah it's true that lower spec computers could struggle with loading ZASF , this particualer machine i am using for this demo is quite adequate .
![specs](https://cloud.githubusercontent.com/assets/335217/6508491/051b2c9a-c351-11e4-97fe-cf37c01ec329.png)

**Comment by tresf** [comment]
> So you are proposing to check if the event prevents the audio from playing cause obviously the specs of a computer would also play a role.

We should review the Show GUI button and see if we're accidentally starting the ZynGUI thread with the wrong priority.

I've never experienced this issue personally.

-Tres

**Comment by mikobuntu** [comment]
@tresf  Great, hopefully this can be fixed and I have always had this problem as far back as I can remember.

**Comment by mikobuntu** [comment]
@unfa Do you experience this behaviour using ZynAdd ?

**Comment by tresf** [comment]
@mikobuntu,

A good place to start is perhaps here, where the slot is setup:

[ZynAddSubFx.cpp#L538](https://github.com/LMMS/lmms/blob/master/plugins/zynaddsubfx/ZynAddSubFx.cpp#L538)

I would recommend checking the thread priority in `toggleUI()` here: [ZynAddSubFx.cpp#L633](https://github.com/LMMS/lmms/blob/master/plugins/zynaddsubfx/ZynAddSubFx.cpp#L633) to see if it is firing the GUI show event with an abnormally high priority.

If it does, we should experiment switching it to use `QThread::IdlePriority`, `QThread::LowPriority` or `QThread::LowestPriority`.

**Comment by Umcaruje** [comment]
I can confirm that this has happened to me almost every time I open the Zyn GUI.

**Comment by musikBear** [comment]
on win 32 xp also true -all previous versions of lmms, but the 'sound dropout' is perhaps not really an issue of significans as such, *unless* it is a sympton of lmms having a brief 'cricis', whitch in extreme circumstances, could lead to a crash, which very well could be a priority thing :+1: for that thought. If thats not the case, imo it is not a serious issue, for 'normal' use. Obviously this could have implications for live-perfomance usage.

**Comment by badosu** [comment]
Well, I wonder if this is ctually possible for LMMS to fix, looks like the GUI interfering with the sound processing of ZASFX is an issue with itself.

The most recent version released had some really nice fixes that can include this, see: http://libremusicproduction.com/news/20150226-zynaddsubfx-250-now-available.

**Comment by PhysSong** [comment]
Confirmed. Since toggling GUI locks `m_pluginMutex`, it blocks whole audio processing, which is definitely undesired. Launching and initializing remote plugin may take long time and cause audio dropout.
I suggest not to call `QMutex::lock` in `play`. I think `tryLock` should replace it and no sound should played if locking fails. This may apply to VST as well.

**Comment by musikBear** [comment]
> This may apply to VST as well.

👍

**Comment by luzpaz** [comment]
Links to the plugin code are 404 in https://github.com/LMMS/lmms/issues/1825#issuecomment-77396674

**Comment by tresf** [comment]
The refactor moved the file and there's been changes since: https://github.com/LMMS/lmms/blob/master/plugins/ZynAddSubFx/ZynAddSubFx.cpp
